hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/MkdirOperation.java

```diff
@@ -98,14 +107,13 @@ public Boolean execute() throws IOException {
     }
 
     // get the file status of the path.
-    // this is done even for a magic path, to avoid always  issuing PUT
-    // requests. Doing that without a check wouild seem to be an
-    // optimization, but it is not because
-    // 1. PUT is slower than HEAD
-    // 2. Write capacity is less than read capacity on a shard
-    // 3. It adds needless entries in versioned buckets, slowing
-    //    down subsequent operations.
-    FileStatus fileStatus = getPathStatusExpectingDir(dir);
+    // this is not done for magic path i.e. performanceCreation mode.
+    // For performanceCreation mode, we would probe for HEAD only.
+    // For non-performance or regular mode, the probe for both HEAD and LIST would
+    // be done.
+    S3AFileStatus fileStatus = performanceCreation
+        ? probePathStatusOrNull(dir, StatusProbeEnum.HEAD_ONLY)
```

Comment:
this will trigger a needless PUT if there isn't a marker, just children, which hits all the problems in the comments of the existing code.

afrid we need to revert to the old code.